### PR TITLE
[http] Add optional TLS client cert authentication

### PR DESCRIPTION
**Pull request by riton**
##### Checklist

- [x] unit and/or integration tests are included (if applicable)
- [x] documentation is changed or added (if applicable)

### Description of change

This P.R adds ability to authenticate using a _client certificate_ in `http` test suites.

Example:

```yaml
---
http:
  https://cc.in2p3.fr/server/that/requires/tls-client-authn:
    status: 200
    client-cert-auth:
      cert: '/path/to/certificate.pem'
      private-key: '/path/to/private-key.pem'
    [...]
```

### CODE REVIEW

**Comment by riton** [comment]
I've just realized that my editor ( _V.S Code_ ) has updated the _Table Of Content_ of the `manual.md` file behind my back.

If this is a problem for you, let me know and I'll revert those automatic changes.

**Comment by riton** [comment]
Not interested in this feature :innocent: ?

**Comment by ripienaar** [comment]
> Not interested in this feature 😇 ?

The author is in a position where he can't reply atm to these things, it does not mean there's no interest.

File: resource/http.go [hunk 1]
```diff
@@ -7,24 +7,30 @@ import (
 	"github.com/aelsabbahy/goss/util"
 )
 
+type clientCertAuth struct {
+	Cert       string `json:"cert,omitempty" yaml:"cert,omitempty"`
+	PrivateKey string `json:"private-key,omitempty" yaml:"private-key,omitempty"`
```
Comment (ripienaar) [inline resource/http.go:12]: Should probably also accept the CA in case its a full verified connection?

**Comment by aelsabbahy** [comment]
Thank you, this has been merged (with minor changes) and will be in the next goss release.

**Pull request merged**